tests/json/test_load_from_json.py

```diff
@@ -311,48 +408,64 @@ def test_GIVEN_component_with_nx_class_WHEN_loading_from_json_THEN_new_model_con
     assert json_reader.entry.instrument.component_list[1].nx_class == component_class
 
 
-def test_GIVEN_transformation_with_matching_name_WHEN_finding_transformation_by_name_THEN_transformation_is_returned(
-    json_reader, component_with_transformation
+def test_GIVEN_json_with_component_depending_on_transfrom_WHEN_loaded_THEN_component_in_model_contains_transform(
+    json_dict_with_component_and_transform, json_reader
 ):
-    transformation = component_with_transformation.transforms[0]
-    assert transformation == json_reader._get_transformation_by_name(
-        component_with_transformation, transformation.name, "DependentComponentName"
-    )
+    json_reader._load_from_json_dict(json_dict_with_component_and_transform)
+    for component in json_reader.entry.instrument.component_list:
+        if component.name == "test_component":
+            assert len(component.transforms) == 1
+            assert component.transforms[0].name == "location"
 
 
-def test_GIVEN_no_transformation_with_matching_name_WHEN_finding_transformation_by_name_THEN_warning_message_is_created(
-    json_reader, component_with_transformation
+def test_GIVEN_json_with_component_depending_on_non_existent_transfrom_WHEN_loaded_THEN_warning_is_added(
+    json_dict_with_component, json_reader
 ):
-    n_warnings = len(json_reader.warnings)
+    depends_on_dataset_str = """
+    {
+      "name":"depends_on",
+      "type":"dataset",
+      "attributes":[],
+      "dataset":{
+        "type":"string",
+        "size":"1"
+      },
+      "values": "/entry/instrument/test_component/transformations/location"
+    }
+    """
+    depends_on_dataset = json.loads(depends_on_dataset_str)
 
-    transformation_name = component_with_transformation.transforms[0].name
-    dependent_component_name = "DependentComponentName"
+    # Add depends_on dataset which points to a transformation which does not exist in the JSON
+    json_dict_with_component["children"][0]["children"][0]["children"][0][
+        "children"
+    ].append(depends_on_dataset)
+    json_reader._load_from_json_dict(json_dict_with_component)
+
+    assert len(json_reader.warnings) > 0
```

Comment:
Same thing here, this test will too easily fail silently (to test for what it is supposed to). They way I see it there are two options:
1. Modify the test to check that you get the the warning you expect.
2. Make another test that has the minimum difference required not to trigger the warning and that thus will not produce any warning messages.

I prefer #1 but #2 is acceptable.

Reply:
I've tried to address this by swapping warning strings out for warning objects which can be type checked in the tests. Hopefully this is an acceptable compromise between the test being too lax and too brittle.